object_store/src/options.rs

```diff
@@ -0,0 +1,129 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::collections::HashMap;
+pub use std::str::FromStr;
+
+#[cfg(any(feature = "gcp", feature = "aws", feature = "azure", feature = "http"))]
+use crate::client::ClientOptions;
+
+#[cfg(any(feature = "aws"))]
+use crate::aws::AmazonS3ConfigKey;
+#[cfg(feature = "azure")]
+use crate::azure::AzureConfigKey;
+#[cfg(feature = "gcp")]
+use crate::gcp::GoogleConfigKey;
+
+/// Options used for configuring backend store
+#[derive(Clone, Debug, Default)]
+pub struct StoreOptions {
+    /// Store specific options like key, secret, region etc.
+    pub store_options: HashMap<String, String>,
```

Comment:
```suggestion
    store_options: HashMap<String, String>,
```

In case we want to extend this further later

Reply:
done.